apollo-client/src/main/java/com/ctrip/framework/apollo/spring/util/BeanRegistrationUtil.java

```diff
@@ -48,9 +57,9 @@ public static boolean registerBeanDefinitionIfNotExists(BeanDefinitionRegistry r
         return false;
       }
 
-      if (beanDefinition instanceof AnnotatedBeanDefinition) {
-        MethodMetadata metadata = ((AnnotatedBeanDefinition) beanDefinition).getFactoryMethodMetadata();
-        if (metadata != null && Objects.equals(metadata.getReturnTypeName(), beanClass.getName())) {
+      if (BD_MAPPING.containsKey(beanClass.getName()) && beanDefinition.getSource() != null && beanDefinition.getSource() instanceof MethodMetadata) {
```

Comment:
It seems we could execute `BD_MAPPING.containsKey(beanClass.getName())` and `BD_MAPPING.get(beanClass.getName())` outside the for loop?

Reply:
oh, maybe we should check every beanDefinition, and it should be leave as this? @nobodyiam

Reply:
Yes, we need to check every bean definition, but I think it's not necessary to execute `BD_MAPPING.containsKey(beanClass.getName())` and `BD_MAPPING.get(beanClass.getName())` everytime in the for loop, as they always return the same result.

```java
String xxx = BD_MAPPING.get(beanClass.getName());
for (String candidate : candidates) {
  ...
}
```

But it's also okay to leave it as it is now since the operations are not expensive.